src/VisualStudio/Core/Impl/Options/OptionBinding.cs

```diff
@@ -1,34 +1,41 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System.ComponentModel;
 using Microsoft.CodeAnalysis.Options;
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.Options
 {
-    internal class OptionBinding<T>
+    internal class OptionBinding<T> : INotifyPropertyChanged
     {
-        private readonly IOptionService _optionService;
+        private readonly OptionStore _optionStore;
         private readonly Option<T> _key;
 
-        public OptionBinding(IOptionService optionService, Option<T> key)
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        public OptionBinding(OptionStore optionStore, Option<T> key)
         {
-            _optionService = optionService;
+            _optionStore = optionStore;
             _key = key;
+
+            _optionStore.OptionChanged += (sender, e) =>
+            {
+                if (e.Option == _key)
+                {
+                    PropertyChanged?.Raise(this, new PropertyChangedEventArgs(nameof(Value)));
```

Comment:
Are we raising `PropertyChanged` here because option store setting isn't synchronous? Otherwise we could just raise in the setter

Reply:
oh! Nm, the value could change outside of this binding. I get it now :)